backend/src/main/java/com/selflearn/backend/security/dtos/SignupDto.java

```diff
@@ -0,0 +1,17 @@
+package com.selflearn.backend.security.dtos;
+
+import com.selflearn.backend.security.validators.constraints.PasswordsEqualConstraint;
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.Pattern;
+import lombok.Getter;
+
+@Getter
+@PasswordsEqualConstraint(message = "Passwords do not match")
+public class SignupDto {
```

Comment:
consider using Record instead of Class for immutable class
https://www.baeldung.com/java-record-keyword